### PR TITLE
allow Matrices in transformations of GroupedDataFrame

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/2781

### CODE REVIEW

**Review by pdeffebach — COMMENTED** [review]
It looks like all the action with `select(df::AbstractDataFrame...)` is happening inside `manipulate` [here](https://github.com/JuliaData/DataFrames.jl/blob/a671a3b86537a253d461ff54b87ae2c26e4e8a04/src/abstractdataframe/selection.jl#L1288). 

Why not use the same structure / logic with SplitApplyCombine?

**Comment by bkamins** [comment]
We essentially use the same structure just the function is not called `manipulate`, but `_combine_prepare` and then super complex logic.

Originally I preferred to add type assertions in `select` etc. in `GroupedDataFrame` case to be on a safe side - if I am not able to write a correct signature then it means that the design is too complex :) (and here what happened is that we have added support for matrices, but forgotten to add signatures).

**Comment by pdeffebach** [comment]
Okay. It sounds like we might want to re-factor in the future, but shouldn't use this PR to do it.

**Comment by bkamins** [comment]
Yes - refactoring is something I wanted to do for a long time, but you know how it is: we add features, the code gest more complex, at some point it is super hard to change anything unless you rewrite it from scratch (that is why I put so much attention to tests and documentation).

File: src/groupeddataframe/splitapplycombine.jl [hunk 1]
```diff
@@ -27,7 +27,8 @@ function _combine_prepare(gd::GroupedDataFrame,
                           keepkeys::Bool, ungroup::Bool, copycols::Bool,
                           keeprows::Bool, renamecols::Bool)
     for cei in cs
-        @assert cei isa Union{Pair, Base.Callable, ColumnIndex, MultiColumnIndex}
+        @assert cei isa Union{Pair, Base.Callable, ColumnIndex, MultiColumnIndex,
+                              AbstractVecOrMat{<:Pair}}
```
Comment (nalimilan) [inline src/groupeddataframe/splitapplycombine.jl:31]: How were vectors supported before this PR?

Comment (bkamins) [reply src/groupeddataframe/splitapplycombine.jl:31]: I could add `AbstractMatrix{<:Pair}` in the union, but I thought that it is cleaner to use `AbstractVecOrMat{<:Pair}` to signal the design intention.

Comment (bkamins) [reply src/groupeddataframe/splitapplycombine.jl:31]: @nalimilan - do we agree on this approach?

Comment (bkamins) [reply src/groupeddataframe/splitapplycombine.jl:31]: @nalimilan - ok to merge?

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**